### ISSUE TITLE
(Bug report) assetPath issue on upgrade from v0.53.2 to v0.59.3

### ISSUE THREAD

**Issue by jat255** [zadam/trilium#3797]
### Trilium Version

v0.59.3

### What operating system are you using?

Other Linux

### What is your setup?

Server access only

### Operating System Version

Alpine Linux v3.16

### Description

I was previously using v0.53.2 without issue and attempted to upgrade to v0.59.3 today. I have Trilium installed in an alpine linux LXC container on Proxmox. The installation appeared to succeed with no issue, updating to the latest version via `git fetch` and `git checkout`, and then running `npm install` and `npm rebuild`. Startup of the app appeared fine.

When I attempt to navigate to my server in the web browser, I see the following in the browser:

```json
{
  "message":"/usr/src/app/trilium/src/views/desktop.ejs:38\n    36|         isProtectedSessionAvailable: <%= isProtectedSessionAvailable %>,\n    37|         triliumVersion: \"<%= triliumVersion %>\",\n >> 38|         assetPath: \"<%= assetPath %>\",\n    39|         appPath: \"<%= appPath %>\",\n    40|         TRILIUM_SAFE_MODE: <%= !!process.env.TRILIUM_SAFE_MODE %>\n    41|     };\n\nassetPath is not defined"
}
``` 

And the following in the Trilium server log file:

```
Generated CSRF token <redacted> with secret undefined
ReferenceError: /usr/src/app/trilium/src/views/desktop.ejs:38
    36|         isProtectedSessionAvailable: <%= isProtectedSessionAvailable %>,
    37|         triliumVersion: "<%= triliumVersion %>",
 >> 38|         assetPath: "<%= assetPath %>",
    39|         appPath: "<%= appPath %>",
    40|         TRILIUM_SAFE_MODE: <%= !!process.env.TRILIUM_SAFE_MODE %>
    41|     };

assetPath is not defined
    at eval ("/usr/src/app/trilium/src/views/desktop.ejs":45:26)
    at desktop (/usr/src/app/trilium/node_modules/ejs/lib/ejs.js:703:17)
    at tryHandleCache (/usr/src/app/trilium/node_modules/ejs/lib/ejs.js:274:36)
    at View.exports.renderFile [as engine] (/usr/src/app/trilium/node_modules/ejs/lib/ejs.js:491:10)
    at View.render (/usr/src/app/trilium/node_modules/express/lib/view.js:135:8)
    at tryRender (/usr/src/app/trilium/node_modules/express/lib/application.js:657:10)
    at Function.render (/usr/src/app/trilium/node_modules/express/lib/application.js:609:3)
    at ServerResponse.render (/usr/src/app/trilium/node_modules/express/lib/response.js:1039:7)
    at index (/usr/src/app/trilium/src/routes/index.js:23:9)
    at Function.cb (/usr/src/app/trilium/src/routes/routes.js:159:34) {
  path: '/usr/src/app/trilium/src/views/desktop.ejs'
}
```

I should add that my Trilium server is behind a reverse proxy (which was working fine on v0.53.2), but I see the same error when attempting to navigate directly via the server's IP as well.

### Error logs

Here is what I see in the app startup log the first time I restarted after the version upgrade. I don't see anything unusual there, although perhaps `"assetPath"` should be in that configuration blurb?

```
App db version is 213, while db version is 196. Migration needed.
DB size: 250155 KB
Trusted reverse proxy: false
App HTTP server starting up at port 80
{
  "appVersion": "0.59.3",
  "dbVersion": 213,
  "syncVersion": 29,
  "buildDate": "2023-03-27T23:06:14+02:00",
  "buildRevision": "9881e6de3e4966af39ec6245562dca6ac7b25eaa",
  "dataDirectory": "/data",
  "clipperProtocolVersion": "1.0",
  "utcDateTime": "2023-04-03T17:16:32.061Z"
}
CPU model: AMD Turion(tm) II Neo N54L Dual-Core Processor, logical cores: 1 freq: 2200 Mhz
Port 80 requires elevated privileges. It's recommended to use port above 1024.
```

EDIT: comparing to prior upgrades, it looks like perhaps the database migration is not completing, because I tried stopping and starting Trilium again, and it says the DB version is still 196. I also am not seeing in the `Migration to version XXX has been successful.` message, so I'll wait a while and see if it finishes. It seems to be working on the database backup, I believe, so the app hasn't actually fully started.

**Comment by jat255** [comment]
Docker's not involved. I'm running in an LXC container via Proxmox, but my install is "bare metal" in the sense that I ssh into the container and then configure everything there, following this procedure: https://github.com/zadam/trilium/wiki/Manual-server-installation

**Comment by jat255** [comment]
I've gotten a bit further with this (I completely redownloaded and re-ran the install and rebuild commands), and was able to get the application started and I could login to my existing data directory, but when I try to load a page, I'm seeing the following error:

```
Generated CSRF token BV1FCuJS-YOFN0ZlfKMk7YkmRaquoKJsHXG8 with secret undefined
Error: Router not found for request /assets/v0.59.3/app-dist/desktop.js
    at /home/node/trilium/src/app.js:94:17
    at Layer.handle [as handle_request] (/home/node/trilium/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/node/trilium/node_modules/express/lib/router/index.js:328:13)
    at /home/node/trilium/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/home/node/trilium/node_modules/express/lib/router/index.js:346:12)
    at next (/home/node/trilium/node_modules/express/lib/router/index.js:280:10)
    at Layer.handle [as handle_request] (/home/node/trilium/node_modules/express/lib/router/layer.js:91:12)
    at trim_prefix (/home/node/trilium/node_modules/express/lib/router/index.js:328:13)
    at /home/node/trilium/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/home/node/trilium/node_modules/express/lib/router/index.js:346:12) {
  status: 404
}
```

Any ideas what might cause this?

**Comment by jat255** [comment]
Yes, the above issue was the problem. I followed what's in the `Dockerfile` manually, and this got me a working distribution. I would recommend that the https://github.com/zadam/trilium/wiki/Manual-server-installation wiki page be updated to reflect the new requirements.

I attempted to use the packaged server installation method as well, but that did not work because Alpine does not use glibc, so I had to compile manually to get it to work. All is well now. Thanks for your help.
